src/hotspot/share/jfr/leakprofiler/chains/edgeUtils.cpp

```diff
@@ -72,7 +72,7 @@ const Symbol* EdgeUtils::field_name(const Edge& edge, jshort* modifiers) {
     JavaFieldStream jfs(ik);
     while (!jfs.done()) {
       if (offset == jfs.offset()) {
-        *modifiers = jfs.access_flags().as_short();
+        *modifiers = jfs.access_flags().as_int();
```

Comment:
This looks wrong - we want a short and you extracted as an int when it was already a short. ??

Reply:
I changed as_int() to as_unsigned_short() which hopefully is less confusing, so resolving these conversations/questions.